agent/config/builder.go

```diff
@@ -2528,17 +2528,20 @@ func validateAutoConfigAuthorizer(rt RuntimeConfig) error {
 }
 
 func (b *builder) cloudConfigVal(v *CloudConfigRaw) (val hcpconfig.CloudConfig) {
+	val.ResourceID = os.Getenv("HCP_RESOURCE_ID")
```

Comment:
I think you will need to use `LookupEnv` here similar to [what the SDK uses](https://github.com/hashicorp/hcp-sdk-go/blob/dbc8bfc75af83d596ca8d33705c466f1ef43708f/config/env.go#L67) or else this will set an empty string if it is null.  Previous iterations of the SDK also checked for null rather than empty string and would fail.  So, probably best to only set this when the env var is actually configured.

Reply:
The `val` variable here is already a zero-value of `hcp.CloudConfig` because of the named return, so if `HCP_RESOURCE_ID` is not set then it stays empty. The SDK is checking that it's set first because that function is mutating potentially non-empty config. 

I updated this to make the `val` variable declaration clearer. This isn't really a case to use named returns: [style guide](https://eng-handbook.hashicorp.services/language-style/go/#return-values).